ddtrace/contrib/django/patch.py

```diff
@@ -82,7 +89,10 @@ def cursor(django, pin, func, instance, args, kwargs):
             "django.db.alias": alias,
         }
         pin = Pin(service, tags=tags, tracer=pin.tracer, app=prefix)
-        return dbapi.TracedCursor(func(*args, **kwargs), pin, config.django)
+        traced_cursor_cls = (
+            Psycopg2TracedCursor if vendor == "postgresql" and Psycopg2TracedCursor is not None else dbapi.TracedCursor
```

Comment:
is vendor guaranteed to be `"postgresql"` for psycopg cursors? 

would we be able to do something like `isinstance(cursor, psycopg.cursor.Cursor)` to be more certain?

Reply:
The vendor attribute on the postgresql database wrapper has been `postgres` for quite some time, even with the older module that was deprecated in 2.0: https://github.com/django/django/blob/stable/1.8.x/django/db/backends/postgresql_psycopg2/base.py#L68

But, I do like your suggestion to check the cursor we get from the wrapped func call and wrap accordingly.

Reply:
👍🏻 agree this might be too brittle, especially if a different postgresql client were used